scripts/suite_objects.py

```diff
@@ -2157,26 +2265,83 @@ def write(self, outfile, host_arglist, indent, const_mod,
             group_type = 'run'      # Allocate for entire run
         # end if
         # Collect information on local variables
-        subpart_vars = {}
+        subpart_allocate_vars = {}
+        subpart_optional_vars = {}
+        subpart_scalar_vars = {}
         allocatable_var_set = set()
+        optional_var_set = set()
+        pointer_var_set = list()
+        inactive_var_set = set()
         for item in [self]:# + self.parts:
             for var in item.declarations():
                 lname = var.get_prop_value('local_name')
-                if lname in subpart_vars:
-                    if subpart_vars[lname][0].compatible(var, self.run_env):
+                sname = var.get_prop_value('standard_name')
+                if (lname in subpart_allocate_vars) or (lname in subpart_optional_vars) or (lname in subpart_scalar_vars):
+                    if subpart_allocate_vars[lname][0].compatible(var, self.run_env):
                         pass # We already are going to declare this variable
                     else:
                         errmsg = "Duplicate Group variable, {}"
                         raise ParseInternalError(errmsg.format(lname))
                     # end if
                 else:
-                    subpart_vars[lname] = (var, item)
+                    opt_var = var.get_prop_value('optional')
                     dims = var.get_dimensions()
                     if (dims is not None) and dims:
-                        allocatable_var_set.add(lname)
+                        if opt_var:
+                            if (self.call_list.find_variable(standard_name=sname)):
+                                subpart_optional_vars[lname] = (var, item, opt_var)
+                                optional_var_set.add(lname)
+                            else:
+                                inactive_var_set.add(var)
```

Comment:
being picky - can you add an "# end if" here?

Reply:
done.